### PR TITLE
[Gutenberg] cancel upload on image block removal

### PR DESCRIPTION
**Pull request by mzorz**
This PR implements media upload cancellation for Gutenberg deleted blocks.

Gutenberg PR https://github.com/WordPress/gutenberg/pull/14391
Gutenberg mobile PR https://github.com/wordpress-mobile/gutenberg-mobile/pull/736

To test:
1. start a draft
2. insert an image block
3. choose an image from the device so it starts uploading
4. once it starts uploading, tap on the block's trash icon
5. observe the foreground upload notification is also dismissed (meaning, the upload has been cancelled).

Test the same on Aztec (to delete an image, just place the cursor to the right of an image and hit backspace) and verify the behavior described in step 5 above.

Update release notes:

- [x] If there are user facing changes, I have added an item to `RELEASE-NOTES.txt`.

### CODE REVIEW

File: WordPress/src/main/java/org/wordpress/android/ui/posts/EditPostActivity.java [hunk 1]
```diff
@@ -3482,9 +3482,9 @@ public void onMediaUploadCancelClicked(String localMediaId) {
     @Override
     public void onMediaDeleted(String localMediaId) {
         if (!TextUtils.isEmpty(localMediaId)) {
-            if (mShowAztecEditor) {
-                mAztecBackspaceDeletedMediaItemIds.add(localMediaId);
-                UploadService.setDeletedMediaItemIds(mAztecBackspaceDeletedMediaItemIds);
+            if (mShowAztecEditor && !mShowGutenbergEditor) {
+                mAztecBackspaceDeletedOrGbBlockDeletedMediaItemIds.add(localMediaId);
```
Comment (marecar3) [inline WordPress/src/main/java/org/wordpress/android/ui/posts/EditPostActivity.java]: Can we put those 2 lines of code in some method so we can call it from both blocks of code to order to increase reusability?

File: WordPress/src/main/java/org/wordpress/android/ui/posts/EditPostActivity.java [hunk 2]
```diff
@@ -3493,6 +3493,9 @@ public void onMediaDeleted(String localMediaId) {
                     return;
                 }
 
+                mAztecBackspaceDeletedOrGbBlockDeletedMediaItemIds.add(localMediaId);
```
Comment (marecar3) [inline WordPress/src/main/java/org/wordpress/android/ui/posts/EditPostActivity.java]: cc:reuse

**Comment by marecar3** [comment]
Hey @mzorz, can we refer to the GB mobile PR, so that we can easily find the corresponding PR?

**Comment by mzorz** [comment]
> Hey @mzorz, can we refer to the GB mobile PR, so that we can easily find the corresponding PR?

They were already linked, but updated the PR description to include those links now 👍

File: WordPress/src/main/java/org/wordpress/android/ui/posts/EditPostActivity.java (hunk 1 above)
Comment (mzorz) [reply WordPress/src/main/java/org/wordpress/android/ui/posts/EditPostActivity.java]: 👍 addressed in 133db73583

File: WordPress/src/main/java/org/wordpress/android/ui/posts/EditPostActivity.java (hunk 2 above)
Comment (mzorz) [reply WordPress/src/main/java/org/wordpress/android/ui/posts/EditPostActivity.java]: addressed in 133db73583 (same as above) 👍

**Pull request merged**